software/spmd/bsg_cuda_lite_runtime/dram_device_allocated/kernel_dram_device_allocated.c

```diff
@@ -23,5 +26,10 @@ int  __attribute__ ((noinline)) kernel_dram_device_allocated(int *addr, int bloc
 
 	bsg_tile_group_barrier(&r_barrier, &c_barrier); 
 
+	if (__bsg_id == 0)
+		bsg_print_stat(1000 + __bsg_tile_group_id);
```

Comment:
What's the purpose of 1000? It seems like a magic number

Reply:
Oh it is a magic number, but I don't have a better way of doing things at the moment. 
You see, with each bsg_print_stats() instruction, a line of stats is printed out for every tile in the manycore, even with the "if (__bsg_id =0)" condition.
And I need two of these print_stats, one at the beginning and one at the end of kernel, so I can subtract the cycles and calculate the total execution time, and I need a way to distinct between the two statements using the tag, but the tag should also indicate which tile group the statement is coming from, since multiple tile groups might use the same tile one after another. 
So basically statements with tags <1000 are beginning statements, and those with tag >1000 are end statements. Do you have any ideas how I can do this without using a magic number? 
I guess I can use positive and negative tags, how does that sound?

Reply:
Actually negative and positive wouldn't work for tile group 0. So I'm not sure if there is a better way to do this, do you have any ideas?

Reply:
I think it's fine to pick whatever value for the print_stat_tag, as long as it's easily identifiable.

Reply:
I don't mind the number, I would rather have it be defined as a macro. 

And I think it would be better represented as a hex number, so that the binary interpretation (if ever needed) is easily distinguishable

Reply:
yup, macro would be better. I used 0xdead in the past.